real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/HillCrestImuNode.java

```diff
@@ -42,12 +39,19 @@ public HillCrestImuNode() {
         super();
         Discovery discover = Discovery.getInstance();
         discover.addListener(this);
-
+        registeredDevices = new ArrayList<>();
```

Comment:
I think this should be before we add this class as a listener, you may end up with a race condition where we try to add to the list before we've initialized it